src/main/java/org/gradle/profiler/mutations/ApplyProjectDependencyChangeMutator.java

```diff
@@ -123,5 +139,78 @@ public ProjectCombinations(List<String> projectNames, Set<Set<String>> combinati
         public Set<String> getNextCombination() {
             return combinations.next();
         }
+
+        public static ProjectCombinations createProjectCombinations(int numberOfRequiredCombinations, int appliedProjectsCount) {
```

Comment:
No need for this to be `public` (we typically use default visibility + `@VisibleForTesing` for methods that need to be available from tests). The methods can also be part of `ApplyProjectDependencyChangeMutator` itself, though that might be a question of taste.

Reply:
Removed public and add annotation